app/src/main/java/fr/free/nrw/commons/nearby/mvp/fragments/NearbyParentFragment.java

```diff
@@ -0,0 +1,842 @@
+    @BindView(R.id.bookmarkButton) LinearLayout bookmarkButton;
+    @BindView(R.id.wikipediaButton) LinearLayout wikipediaButton;
+    @BindView(R.id.wikidataButton) LinearLayout wikidataButton;
+    @BindView(R.id.directionsButton) LinearLayout directionsButton;
+    @BindView(R.id.commonsButton) LinearLayout commonsButton;
+    @BindView(R.id.description) TextView description;
+    @BindView(R.id.title) TextView title;
+    @BindView(R.id.category) TextView distance;
+    @BindView(R.id.icon) ImageView icon;
+    @BindView(R.id.search_this_area_button) Button searchThisAreaButton;
+    @BindView(R.id.map_progress_bar) ProgressBar progressBar;
+    @BindView(R.id.container_sheet) FrameLayout frameLayout;
+    @BindView(R.id.loading_nearby_list) ConstraintLayout loadingNearbyLayout;
+
+    @Inject LocationServiceManager locationManager;
+    @Inject NearbyController nearbyController;
+    @Inject @Named("default_preferences") JsonKvStore applicationKvStore;
+    @Inject BookmarkLocationsDao bookmarkLocationDao;
+    @Inject ContributionController controller;
+    @Inject WikidataEditListener wikidataEditListener;
+
+    private BottomSheetBehavior bottomSheetListBehavior;
+    private BottomSheetBehavior bottomSheetDetailsBehavior;
+    private Animation rotate_backward;
+    private Animation fab_close;
+    private Animation fab_open;
+    private Animation rotate_forward;
+
+    private static final double ZOOM_LEVEL = 14f;
+    private final String NETWORK_INTENT_ACTION = "android.net.conn.CONNECTIVITY_CHANGE";
+    private BroadcastReceiver broadcastReceiver;
+    private boolean isNetworkErrorOccurred = false;
+    private Snackbar snackbar;
+    private View view;
+    public NearbyParentFragmentPresenter nearbyParentFragmentPresenter;
+    private boolean isDarkTheme;
+    private boolean isFABsExpanded;
+    private Marker selectedMarker;
+    private Place selectedPlace;
+
+    private final double CAMERA_TARGET_SHIFT_FACTOR_PORTRAIT = 0.005;
+    private final double CAMERA_TARGET_SHIFT_FACTOR_LANDSCAPE = 0.004;
+
+    private NearbyMapFragment nearbyMapFragment;
+    private NearbyListFragment nearbyListFragment;
+
+    public static final String TAG_RETAINED_MAP_FRAGMENT = NearbyMapFragment.class.getSimpleName();
+    private static final String TAG_RETAINED_LIST_FRAGMENT = NearbyListFragment.class.getSimpleName();
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        view = inflater.inflate(R.layout.fragment_nearby_parent, container, false);
+        ButterKnife.bind(this, view);
+        // Inflate the layout for this fragment
+        return view;
+
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        nearbyMapFragment = getNearbyMapFragment();
+        nearbyListFragment = getListFragment();
+    }
+
+
+    private void initViews() {
```

Comment:
Break initViews into smaller methods. As of now all UI initialisation is happening in the same method.